src/components/eip4626/User.tsx

```diff
@@ -7,33 +7,80 @@ import { VaultInfo } from "@/components/eip4626/user/VaultInfo";
 import { VaultDeposit } from "@/components/eip4626/user/VaultDeposit";
 import { Eip4626Context } from "@/contexts/Eip4626Context";
 import { VaultAddReward } from "@/components/eip4626/user/VaultAddReward";
+import { MintAssetToken } from "@/components/eip4626/user/MintAssetToken";
+import { useReadHederaVaultAssetQueries } from "@/hooks/eip4626/useReadHederaVaultAsset";
+import { EvmAddress } from "@/types/types";
 
 export default function User() {
-  const [vaultSelected, setVaultSelected] = useState("" as `0x${string}`);
+  const [vaultSelected, setVaultSelected] = useState("" as EvmAddress);
+  const [vaultAssetSelected, setVaultAssetSelected] = useState(
+    "" as EvmAddress,
+  );
 
+  const { deployedProxyHtsTokens } = useContext(Eip4626Context);
   const { deployedVaults } = useContext(Eip4626Context);
 
+  const vaultAddresses = deployedVaults.map((item) => item?.["args"]?.[0]);
+  const vaultAssetsFetched = useReadHederaVaultAssetQueries(vaultAddresses);
+
+  const filteredVaultsBySelectedAsset = vaultAssetsFetched.filter(
+    (item) => item?.data?.vaultAssetAddress === vaultAssetSelected,
+  );
+
+  const filteredVaultsForSelect = deployedVaults.filter((item) => {
+    return filteredVaultsBySelectedAsset.find(
+      (subItem) => subItem?.data?.vaultAddress === item?.["args"]?.[0],
+    );
+  });
+
   return (
     <>
-      <Stack spacing={4} align="center">
-        <Select
-          placeholder="Select vault for operation"
-          onChange={(item) => {
-            const vaultItem = deployedVaults.find(
-              (itemSub) => itemSub?.["args"]?.[0] === item.target.value,
-            );
-            setVaultSelected(vaultItem?.["args"]?.[0]);
-          }}
-          variant="outline"
-        >
-          {deployedVaults.map((item) => (
-            <option key={item?.["args"]?.[0]} value={item?.["args"]?.[0]}>
-              {item?.["args"]?.[1]} ({item?.["args"]?.[2]}) [
-              {item?.["args"]?.[0]}]
-            </option>
-          ))}
-        </Select>
+      {!deployedProxyHtsTokens.length ? (
```

Comment:
I think it's always better to deal with positive conditions like:
`deployedProxyHtsTokens.length ? <do /> : <otherwise>`